pandas/core/groupby/generic.py

```diff
@@ -339,29 +341,99 @@ def _aggregate_multiple_funcs(self, arg, _level):
 
         return DataFrame(results, columns=columns)
 
-    def _wrap_series_output(self, output, index, names=None):
-        """ common agg/transform wrapping logic """
-        output = output[self._selection_name]
+    def _wrap_series_output(
+        self, output: Mapping[int, AnyArrayLike], index: Index, columns: Index
+    ) -> Union[Series, DataFrame]:
+        """
+        Wraps the output of a SeriesGroupBy operation into the expected result.
 
-        if names is not None:
-            return DataFrame(output, index=index, columns=names)
+        Parameters
+        ----------
+        output : dict[int, np.ndarray]
+            Dict where the key represents the columnar-index and the values are
+            the actual results. Must be ordered from 0..n
+        index : pd.Index
+            Index to apply to the output.
+        columns : pd.Index
+            Columns to apply to the output.
+
+        Returns
+        -------
+        Series or DataFrame
+
+        Notes
+        -----
+        In the vast majority of cases output and columns will only contain one
+        element. The exception is operations that expand dimensions, like ohlc.
+        """
+        assert len(output) == len(columns)
+        if len(output) > 1:
+            result = DataFrame(output, index=index)  # type: Union[Series, DataFrame]
+            result.columns = columns
         else:
-            name = self._selection_name
-            if name is None:
-                name = self._selected_obj.name
-            return Series(output, index=index, name=name)
+            result = Series(output[0], index=index, name=columns[0])
+
+        return result
 
-    def _wrap_aggregated_output(self, output, names=None):
+    def _wrap_aggregated_output(
+        self, output: Mapping[int, AnyArrayLike], columns: Index
+    ) -> Union[Series, DataFrame]:
+        """
+        Wraps the output of a SeriesGroupBy aggregation into the expected result.
+
+        Parameters
+        ----------
+        output : dict[int, np.ndarray]
```

Comment:
dict->Mapping?

Reply:
actually, looks like this docstring is really similar to the previous one.  can it be shared?

Reply:
Thought about this but there are differences in the signatures of these functions (see `index` parameter). I don't think worth adding to the diff here for substitution but happy to open as a follow up, unless you feel strongly about this